packages/web3/src/exchange/exchange.ts

```diff
@@ -67,7 +58,54 @@ export function getALPHDepositInfo(tx: Transaction): { targetAddress: Address; d
   return Array.from(result.entries()).map(([key, value]) => ({ targetAddress: key, depositAmount: value }))
 }
 
-// we assume that the tx is a simple transfer tx, i.e. isSimpleALPHTransferTx(tx) == true
+function getInputAddresses(tx: Transaction): Address[] {
+  const inputAddresses: Address[] = []
+  for (const input of tx.unsigned.inputs) {
+    try {
+      const address = getAddressFromUnlockScript(input.unlockScript)
+      if (!inputAddresses.includes(address)) {
+        inputAddresses.push(address)
+      }
+    } catch (error) {
+      throw new TraceableError(`Failed to decode address from unlock script`, error)
+    }
+  }
+  return inputAddresses
+}
+
+export function isTokenTransferTx(tx: Transaction): boolean {
+  return isTransferTx(tx) && checkTokenOutput(tx)
+}
+
+export function getTokenDepositInfo(tx: Transaction): {
```

Comment:
Does this generate information for ALPH if ALPH and tokens are sent in a single transaction?

Reply:
No, it only returns the token info.